icu4c/source/common/locid.cpp

```diff
@@ -302,10 +302,14 @@ struct Locale::Heap : public UMemory {
         copyToArray<&Heap::region>(region, this);
     }
 
-    Heap(const Heap& other)
+    Heap(const Heap& other, UErrorCode& status)
         : variantBegin(other.variantBegin),
           fullName(other.fullName),
           baseName(other.baseName) {
+        if (fullName.isEmpty() != other.fullName.isEmpty() ||
+            baseName.isEmpty() != other.baseName.isEmpty()) {
+            status = U_MEMORY_ALLOCATION_ERROR;
+        }
```

Comment:
Condition-continuation indentation should be larger than body indentation, so that we can tell one from the other.
```suggestion
        if (fullName.isEmpty() != other.fullName.isEmpty() ||
                baseName.isEmpty() != other.baseName.isEmpty()) {
            status = U_MEMORY_ALLOCATION_ERROR;
        }
```

Reply:
This code was formatted by `clang-format` so it's correct according to ICU4C's `.clang-format` and such formatting can be found elsewhere in the code base.

Reply:
That's purely a matter of taste. I personally agree with whoever wrote the current configuration and think it's beneficial for readability to have the statements lined up like they are now, but if you disagree with that you'll have to get the configuration changed for trying to get everyone to manually edit the output of the formatter all the time won't be a productive path forward.

Reply:
> We don't routinely use clang format in ICU,

Some of us do, that's why ICU4C has had a custom `.clang-format` for many years now.

> When it has a bug, we should fix its output.

I fully agree with that, I just don't agree that there's something wrong with this formatting here, I quite like it.

Reply:
I don't accept
```
        if (fullName.isEmpty() != other.fullName.isEmpty() ||
            baseName.isEmpty() != other.baseName.isEmpty()) {
            status = U_MEMORY_ALLOCATION_ERROR;
        }
```

I can accept
```
        if (fullName.isEmpty() != other.fullName.isEmpty() ||
                baseName.isEmpty() != other.baseName.isEmpty()) {
            status = U_MEMORY_ALLOCATION_ERROR;
        }
```
(like in [this example](https://google.github.io/styleguide/cppguide.html#Boolean_Expressions))
or
```
        if (fullName.isEmpty() != other.fullName.isEmpty() ||
            baseName.isEmpty() != other.baseName.isEmpty()
        ) {
            status = U_MEMORY_ALLOCATION_ERROR;
        }
```